### PR TITLE
Gregexpr implementation

### PR DESCRIPTION
**Pull request by psolaimani**

### CODE REVIEW

File: core/src/main/java/org/renjin/primitives/text/Text.java [hunk 1]
```diff
@@ -482,12 +482,49 @@ public static IntVector regexpr(String pattern, StringVector vector, boolean ign
...
+                                  boolean fixed, boolean useBytes) {
+
+    ListVector.Builder regexpResults = new ListVector.Builder(0, vector.length());
+    RE re = REFactory.compile(pattern, ignoreCase,  perl, fixed, useBytes);
+
+    for(String text : vector) {
+      IntArrayVector.Builder position = IntArrayVector.Builder.withInitialCapacity(vector.length());
+      IntArrayVector.Builder matchLength = IntArrayVector.Builder.withInitialCapacity(vector.length());
+      int offsetSearch = 0;
+      boolean moreMatch = re.match(text.substring(offsetSearch));
```
Comment (akbertram) [inline core/src/main/java/org/renjin/primitives/text/Text.java]: Why not just 
```
while(re.match(text.substring(offsetSearch)) {
```
?

File: core/src/main/java/org/renjin/primitives/text/Text.java [hunk 2]
```diff
@@ -482,12 +482,49 @@ public static IntVector regexpr(String pattern, StringVector vector, boolean ign
...
+      do {
+        if ( moreMatch) {
+          int start = re.getGroupStart(0);
+          int end = re.getGroupEnd(0);
+          position.add(start + 1 + offsetSearch);
+          matchLength.add( end - start);
+          offsetSearch += end;
+          moreMatch = re.match(text.substring(offsetSearch));
+        } else {
+          position.add(-1);
```
Comment (akbertram) [inline core/src/main/java/org/renjin/primitives/text/Text.java]: Can you add a test case that covers this branch?

**Comment by akbertram** [comment]
Looks good! Add a few more test cases for edge cases like `pattern=""`
